### PR TITLE
Use right crypto lib

### LINKED ISSUE #60215: [BUG] salt.utils.crypt can use wrong crypto library
**Description**

In the `salt.crypt` module we use some logic to determine which crypto library we are using. The logic tries M2Crypto, PyCryptodomex, then PyCrypto. The imports in `salt.utils.crypt` do not try to import M2Crypto or PyCryptodomex. This can have negative side effects. I observed code in the `salt.utils.vt.Terminal` stuck after forking because it was calling `Crypto.Random.reinit` via the `salt.utils.crypt.reinit_crypto` even though PyCryptodomex is installed (PyCryptodomex's version of this method is a no-op).

**Setup**

Install both PyCryptdomex and PyCrypto on the same system. Run cloud deployments.

**Expected behavior**

`salt.utils.crypto` sould not try to import PyCrypto when PyCryptodomex is installed.

**Screenshots**

**Versions Report**

This was observed on 3002.5

This is related to #59903

### PR DESCRIPTION
**Pull request by cmcmarrow**
### What does this PR do?
fix and test reinit by making sure right crypto lib is used

### What issues does this PR fix or reference?
Fixes: #60215

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by cmcmarrow** [comment]
https://github.com/saltstack/salt/pull/60526

File: tests/pytests/unit/utils/test_crypt.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+def test_random():
+    # make sure the right liberty is used for random
+    if HAS_CYPTODOME:
+        assert CryptodomeRandom is salt.utils.crypt.Random
+    elif HAS_CRYPTO:
+        assert CryptoRandom is salt.utils.crypt.Random
+
+
+def test_reinit_crypto():
+    # make sure reinit cryptot does not crash
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/test_crypt.py:34]:
```suggestion
    # make sure reinit crypto does not crash
```

File: tests/pytests/unit/utils/test_crypt.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+try:
+    from Crypto import Random as CryptoRandom  # nosec
+
+    HAS_CRYPTO = True
+except ImportError:
+    HAS_CRYPTO = False
+
+
+def test_random():
+    # make sure the right liberty is used for random
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/test_crypt.py:26]:
```suggestion
    # make sure the right library is used for random
```

File: tests/pytests/unit/utils/test_crypt.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        assert CryptoRandom is salt.utils.crypt.Random
+
+
+def test_reinit_crypto():
+    # make sure reinit cryptot does not crash
+    salt.utils.crypt.reinit_crypto()
+
+    # make sure reinit does not crash when no crypt is found
+    with patch("salt.utils.crypt.HAS_CRYPTO", None):
+        with patch("salt.utils.crypt.Random", None):
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/test_crypt.py:39]: This patch is not needed, right?

Comment (s0undt3ch) [reply tests/pytests/unit/utils/test_crypt.py:39]: Because `Random` usage is always behind an `if HAS_CRYPTO`....

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Pull request closed without merging**